File: src/main/java/com/pedigree/ui/Dialogs.java
package com.pedigree.ui;

import javafx.scene.control.Alert;
import javafx.scene.control.ButtonType;
import javafx.stage.FileChooser;
import javafx.stage.Stage;

import java.nio.file.Path;

public class Dialogs {

    public static Path chooseOpenProjectPath() {
        FileChooser fc = new FileChooser();
        fc.setTitle("Open Project");
        fc.getExtensionFilters().add(new FileChooser.ExtensionFilter("Pedigree Project (*.ped)", "*.ped"));
        var file = fc.showOpenDialog(getActiveWindow());
        return file != null ? file.toPath() : null;
    }

    public static Path chooseSaveProjectPath() {
        FileChooser fc = new FileChooser();
        fc.setTitle("Save Project");
        fc.getExtensionFilters().add(new FileChooser.ExtensionFilter("Pedigree Project (*.ped)", "*.ped"));
        var file = fc.showSaveDialog(getActiveWindow());
        return file != null ? file.toPath() : null;
    }

    public static Path chooseOpenGedcomPath() {
        FileChooser fc = new FileChooser();
        fc.setTitle("Import GEDCOM");
        fc.getExtensionFilters().add(new FileChooser.ExtensionFilter("GEDCOM (*.ged)", "*.ged"));
        var file = fc.showOpenDialog(getActiveWindow());
        return file != null ? file.toPath() : null;
    }

    public static Path chooseSaveGedcomPath() {
        FileChooser fc = new FileChooser();
        fc.setTitle("Export GEDCOM");
        fc.getExtensionFilters().add(new FileChooser.ExtensionFilter("GEDCOM (*.ged)", "*.ged"));
        var file = fc.showSaveDialog(getActiveWindow());
        return file != null ? file.toPath() : null;
    }

    public static Path chooseSaveSvgPath() {
        FileChooser fc = new FileChooser();
        fc.setTitle("Export SVG");
        fc.getExtensionFilters().add(new FileChooser.ExtensionFilter("SVG (*.svg)", "*.svg"));
        var file = fc.showSaveDialog(getActiveWindow());
        return file != null ? file.toPath() : null;
    }

    public static Path chooseSaveHtmlPath() {
        FileChooser fc = new FileChooser();
        fc.setTitle("Export HTML");
        fc.getExtensionFilters().add(new FileChooser.ExtensionFilter("HTML (*.html)", "*.html"));
        var file = fc.showSaveDialog(getActiveWindow());
        return file != null ? file.toPath() : null;
    }

    public static Path chooseSaveImagePath() {
        FileChooser fc = new FileChooser();
        fc.setTitle("Export Image");
        fc.getExtensionFilters().addAll(
                new FileChooser.ExtensionFilter("PNG (*.png)", "*.png"),
                new FileChooser.ExtensionFilter("JPEG (*.jpg, *.jpeg)", "*.jpg", "*.jpeg")
        );
        var file = fc.showSaveDialog(getActiveWindow());
        return file != null ? file.toPath() : null;
    }

    public static void showError(String title, String message) {
        Alert alert = new Alert(Alert.AlertType.ERROR, message, ButtonType.CLOSE);
        alert.setTitle(title);
        alert.showAndWait();
    }

    private static Stage getActiveWindow() {
        for (javafx.stage.Window w : javafx.stage.Window.getWindows()) {
            if (w instanceof Stage s && s.isShowing()) return s;
        }
        return null;
    }
}

File: src/main/kotlin/com/family/tree/ui/Dialogs.kt
package com.family.tree.ui

import javafx.scene.control.Alert
import javafx.scene.control.ButtonType
import javafx.stage.FileChooser
import javafx.stage.Stage
import java.nio.file.Path

/**
 * Kotlin version of Dialogs utility. Preserves Java API via @JvmStatic methods.
 */
object Dialogs {
    @JvmStatic
    fun chooseOpenProjectPath(): Path? {
        val fc = FileChooser()
        fc.title = "Open Project"
        fc.extensionFilters.add(FileChooser.ExtensionFilter("Pedigree Project (*.ped)", "*.ped"))
        val file = fc.showOpenDialog(activeWindow)
        return file?.toPath()
    }

    @JvmStatic
    fun chooseSaveProjectPath(): Path? {
        val fc = FileChooser()
        fc.title = "Save Project"
        fc.extensionFilters.add(FileChooser.ExtensionFilter("Pedigree Project (*.ped)", "*.ped"))
        val file = fc.showSaveDialog(activeWindow)
        return file?.toPath()
    }

    @JvmStatic
    fun chooseOpenGedcomPath(): Path? {
        val fc = FileChooser()
        fc.title = "Import GEDCOM"
        fc.extensionFilters.add(FileChooser.ExtensionFilter("GEDCOM (*.ged)", "*.ged"))
        val file = fc.showOpenDialog(activeWindow)
        return file?.toPath()
    }

    @JvmStatic
    fun chooseOpenRelPath(): Path? {
        val fc = FileChooser()
        fc.title = "Import REL"
        fc.extensionFilters.add(FileChooser.ExtensionFilter("Relatives (*.rel)", "*.rel"))
        val file = fc.showOpenDialog(activeWindow)
        return file?.toPath()
    }

    @JvmStatic
    fun chooseSaveGedcomPath(): Path? {
        val fc = FileChooser()
        fc.title = "Export GEDCOM"
        fc.extensionFilters.add(FileChooser.ExtensionFilter("GEDCOM (*.ged)", "*.ged"))
        val file = fc.showSaveDialog(activeWindow)
        return file?.toPath()
    }

    @JvmStatic
    fun chooseSaveSvgPath(): Path? {
        val fc = FileChooser()
        fc.title = "Export SVG"
        fc.extensionFilters.add(FileChooser.ExtensionFilter("SVG (*.svg)", "*.svg"))
        val file = fc.showSaveDialog(activeWindow)
        return file?.toPath()
    }

    @JvmStatic
    fun chooseSaveHtmlPath(): Path? {
        val fc = FileChooser()
        fc.title = "Export HTML"
        fc.extensionFilters.add(FileChooser.ExtensionFilter("HTML (*.html)", "*.html"))
        val file = fc.showSaveDialog(activeWindow)
        return file?.toPath()
    }

    @JvmStatic
    fun chooseSaveImagePath(): Path? {
        val fc = FileChooser()
        fc.title = "Export Image"
        fc.extensionFilters.addAll(
            FileChooser.ExtensionFilter("PNG (*.png)", "*.png"),
            FileChooser.ExtensionFilter("JPEG (*.jpg, *.jpeg)", "*.jpg", "*.jpeg")
        )
        val file = fc.showSaveDialog(activeWindow)
        return file?.toPath()
    }

    @JvmStatic
    fun showError(title: String, message: String?) {
        val alert = Alert(Alert.AlertType.ERROR, message ?: "", ButtonType.CLOSE)
        alert.title = title
        alert.showAndWait()
    }

    private val activeWindow: Stage?
        get() {
            for (w in javafx.stage.Window.getWindows()) {
                if (w is Stage && w.isShowing) return w
            }
            return null
        }
}
